From 21ed786a9fabb1c0d70ff64db572c5f928134fda Mon Sep 17 00:00:00 2001
From: Frank Binns <frank.binns@imgtec.com>
Date: Tue, 16 Sep 2014 09:45:55 +0100
Subject: [PATCH 47/52] dri: set the __DRI_API_OPENGL bit based on max gl
 compat version

This matches similar behaviour for the __DRI_API_OPENGL_CORE bit.
---
 src/mesa/drivers/dri/common/dri_util.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/mesa/drivers/dri/common/dri_util.c b/src/mesa/drivers/dri/common/dri_util.c
index a0422fc..4614455 100644
--- a/src/mesa/drivers/dri/common/dri_util.c
+++ b/src/mesa/drivers/dri/common/dri_util.c
@@ -171,7 +171,9 @@ driCreateNewScreen2(int scrn, int fd,
                                          gl_version_override);
     }
 
-    psp->api_mask = (1 << __DRI_API_OPENGL);
+    psp->api_mask = 0;
+    if (psp->max_gl_compat_version > 0)
+       psp->api_mask |= (1 << __DRI_API_OPENGL);
     if (psp->max_gl_core_version > 0)
        psp->api_mask |= (1 << __DRI_API_OPENGL_CORE);
     if (psp->max_gl_es1_version > 0)
-- 
1.8.5.4.gfdaaaa2

